Remotion/Validation/Core/Attributes/Validation/LengthValidationAttribute.cs

```diff
@@ -40,39 +40,56 @@ public class LengthValidationAttribute : AddingValidationAttributeBase
     /// <param name="maxLength">The maximum number of characters allowed.</param>
     public LengthValidationAttribute (int minLength, int maxLength)
     {
-      _minLength = minLength;
-      _maxLength = maxLength;
-    }
-
-    public int MinLength
-    {
-      get { return _minLength; }
-    }
-
-    public int MaxLength
-    {
-      get { return _maxLength; }
+      MinLength = minLength;
+      MaxLength = maxLength;
     }
 
     protected override IEnumerable<IPropertyValidator> GetValidators (IPropertyInformation property, IValidationMessageFactory validationMessageFactory)
     {
       ArgumentUtility.CheckNotNull("property", property);
       ArgumentUtility.CheckNotNull("validationMessageFactory", validationMessageFactory);
 
-      LengthValidator validator;
+      IPropertyValidator validator;
       if (string.IsNullOrEmpty(ErrorMessage))
       {
         validator = PropertyValidatorFactory.Create(
             property,
-            parameters => new LengthValidator(MinLength, MaxLength, parameters.ValidationMessage),
+            parameters => CreateValidator(parameters.ValidationMessage),
             validationMessageFactory);
       }
       else
       {
-        validator = new LengthValidator(MinLength, MaxLength, new InvariantValidationMessage(ErrorMessage));
+        validator = CreateValidator(new InvariantValidationMessage(ErrorMessage));
       }
 
       return EnumerableUtility.Singleton(validator);
     }
+
+    private IPropertyValidator CreateValidator (ValidationMessage message)
+    {
+      if (MinLength != 0 && MaxLength > MinLength)
```

Comment:
This if else construct might be better suited to a switch expression/statement. Am not sure if it's more readable though, probably needs investigation.